### PR TITLE
fix: `hoodie.trigger()` should accept optional arguments

### LINKED ISSUE #156: Arguments ignored in custom events
From the [docs](http://docs.hood.ie/en/latest/api/client/hoodie.html#hoodie-trigger), the signature of the `hoodie.trigger()` method is:

```js
hoodie.trigger(eventName[, option1, option2, ...])
```

However, the optional arguments are simply ignored in the [implementation](https://github.com/hoodiehq/hoodie-client/blob/7eb0da2a3eb5af7e113b3dd4acc639cafdebbdcb/lib/events.js#L54):

```js
state.emitter.emit(eventName)
```

In effect, when we trigger a custom event with:

```js
hoodie.trigger('name', object);
```

The passed-in `object` will always be `undefined` in the event handler:

```js
hoodie.on('name', (object) => {
  console.log(object); //=> undefined
});
```

### PR DESCRIPTION
**Pull request by gnowoel**
Trying to fix #156.

### CODE REVIEW

File: lib/events.js [hunk 1]
```diff
@@ -48,10 +48,19 @@ function off (state, eventName, handler) {
...
  */
 function trigger (state, eventName) {
-  state.emitter.emit(eventName)
+  var emitter = state.emitter
+  var len = arguments.length
+  var args = Array(len - 1)
+
+  for (var i = 0; i < len - 1; i++) {
+    args[i] = arguments[i + 1]
+  }
```
Comment (gr2m) [inline lib/events.js]: You can simplify the code above with this line

```js
var args = [].slice.call(arguments, 1)
```

Here is a minimal showcase

```js
(function () {
  var args = [].slice.call(arguments, 1)
  console.log(args)
})('eventName', 1, 2, 3)
```

Does that make sense?

File: tests/specs/events.js [hunk 2]
```diff
@@ -129,3 +129,24 @@ test('"trigger" returns an object', function (t) {
...
+  var option3 = 'option3'
+
+  hoodie.on('account:signin', function () {
+    t.equal(arguments[0], option1, 'receives the first argument')
+    t.equal(arguments[1], option2, 'receives the second argument')
+    t.equal(arguments[2], option3, 'receives the third argument')
+  })
+
+  hoodie.trigger('account:signin', option1, option2, option3)
+})
```
Comment (gr2m) [inline tests/specs/events.js]: 👍

**Comment by gnowoel** [comment]
It's much cleaner now :) Thanks for your suggestion!

**Comment by gr2m** [comment]
published as `@hoodie/client@10.1.1` to npm. Does not show up in [releases](https://github.com/hoodiehq/hoodie-client/releases) due to a [problem with greenkeeper-postpublish](https://travis-ci.org/hoodiehq/hoodie-client/jobs/262057035#L923)

**Pull request merged**